app/parse.py

```diff
@@ -15,8 +34,103 @@ class Product:
     num_of_reviews: int
 
 
+class ChromeDriver:
+    def __init__(self) -> None:
+        self._driver = webdriver.Chrome()
+
+    def __enter__(self) -> webdriver.Chrome:
+        return self._driver
+
+    def __exit__(
+            self,
+            exc_type: Optional[Type[BaseException]],
+            exc_val: Optional[BaseException],
+            exc_tb: Optional[TracebackType]
+    ) -> None:
+        self._driver.close()
+
+
+def get_catalog_name(url: str) -> str:
+    return url.split("/")[-2] if url else "home"
+
+
+def parse_single_product(product: WebElement) -> Product:
+    return Product(
+        title=product.find_element(
+            By.CLASS_NAME, "title"
+        ).get_attribute("title"),
+        description=product.find_element(
+            By.CLASS_NAME, "description"
+        ).text,
+        price=float(product.find_element(
+            By.CLASS_NAME, "price"
+        ).text.replace("$", "")),
+        rating=len(product.find_elements(
+            By.TAG_NAME, "span")
+        ),
+        num_of_reviews=int(product.find_element(
+            By.CLASS_NAME, "review-count"
+        ).text.split()[0]),
+    )
+
+
+def get_selenium_products_page(
+        catalog: str,
+        driver: webdriver.Chrome = None
```

Comment:
Ensure that the `driver` parameter is not `None` before using it. You might want to raise an exception or handle this case to prevent runtime errors.